api/httpjson/RPCserver.go

```diff
@@ -292,9 +272,30 @@ func (s *RPCServer) Start() {
 	}
 
 	httpServer.SetKeepAlivesEnabled(config.Parameters.RPCKeepAlivesEnabled)
+	s.httpServer = httpServer
+	listener, err := net.Listen("tcp", s.httpListener)
+	if err != nil {
+		log.Error("net.Listen: ", err.Error())
+		return
+	}
+	go s.httpServer.Serve(listener)
 
-	go httpServer.Serve(listener)
-	go httpServer.Serve(tlsListener)
+	go func(httpsCertReady chan struct{}) {
+		for {
+			select {
+			case <-httpsCertReady:
+				log.Info("https cert received")
+				tlsListener, err := s.initTlsListen()
+				if err != nil {
+					log.Errorf("Https Cert: %v", err.Error())
+					return
+				}
+				s.httpServer.Serve(tlsListener)
```

Comment:
Why not return after `Serve()`? Go routine will be leaked if serve returns error because `httpsCertReady` will not be fired again. Same for wss